File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamuser/AnonymousLogOnDetails.java
package in.dragonbra.javasteam.steam.handlers.steamuser;

import in.dragonbra.javasteam.enums.EOSType;
import in.dragonbra.javasteam.util.Utils;

/**
 * Represents the details required to log into Steam3 as an anonymous user.
 */
public class AnonymousLogOnDetails {

    private int cellID;

    private EOSType clientOSType;

    private String clientLanguage;

    public AnonymousLogOnDetails() {
        clientOSType = Utils.getOSType();
        clientLanguage = "english";
    }

    public int getCellID() {
        return cellID;
    }

    public void setCellID(int cellID) {
        this.cellID = cellID;
    }

    public EOSType getClientOSType() {
        return clientOSType;
    }

    public void setClientOSType(EOSType clientOSType) {
        this.clientOSType = clientOSType;
    }

    public String getClientLanguage() {
        return clientLanguage;
    }

    public void setClientLanguage(String clientLanguage) {
        this.clientLanguage = clientLanguage;
    }
}

File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamuser/AnonymousLogOnDetails.kt
package `in`.dragonbra.javasteam.steam.handlers.steamuser

import `in`.dragonbra.javasteam.enums.EOSType
import `in`.dragonbra.javasteam.util.Utils

/**
 * Represents the details required to log into Steam3 as an anonymous user.
 *
 * @param cellID Gets or sets the CellID.
 * @param clientOSType Gets or sets the client operating system type.
 * @param clientLanguage Gets or sets the client language.
 */
data class AnonymousLogOnDetails(
    var cellID: Int? = null,
    var clientOSType: EOSType = Utils.getOSType(),
    var clientLanguage: String = "english",
)
